build/media-suite_compile.sh

```diff
@@ -1816,11 +1817,12 @@ if { { [[ $ffmpeg != no ]] && enabled vulkan; } || ! mpv_disabled vulkan; } &&
         do_install d3d{kmthk,ukmdt}.h include/
     cd_safe "$(get_first_subdir -f)"
     do_print_progress "Building Vulkan-Loader"
-    CFLAGS+=" -DSTRSAFE_NO_DEPRECATE" do_cmakeinstall -DBUILD_TESTS=OFF -DCMAKE_SYSTEM_NAME=Windows -DUSE_CCACHE=OFF \
-    -DCMAKE_ASM_COMPILER="$(command -v nasm.exe)" -DVULKAN_HEADERS_INSTALL_DIR="$LOCALDESTDIR" \
+    CC=gcc CXX=g++ CFLAGS+=" -DSTRSAFE_NO_DEPRECATE" do_cmakeinstall -DBUILD_TESTS=OFF \
```

Comment:
any reason why we are setting CC and CXX here? Is ccache or clang broken?

As for the using uasm, I think the issue might be that neither [this line](https://github.com/KhronosGroup/Vulkan-Loader/commit/e1c938170bba5879d95bf2d7b57b4d1d68557cd5#diff-ac8626da51dc3b5d11ecb20f761f7344R137) nor cmake's ASM_MASM language will pick up on it, https://github.com/Kitware/CMake/blob/master/Modules/CMakeDetermineASM_MASMCompiler.cmake

We might need to force set it through `-DCMAKE_ASM_MASM_COMPILER="$(command -v uasm)"`

Reply:
It was broken with the previous check using `COMMAND ${CMAKE_C_COMPILER} --version OUTPUT_VARIABLE COMPILER_VERSION_OUTPUT` but I refactored it to use `CMAKE_SIZEOF_VOID_P STREQUAL "8"` instead, so it's probably not needed anymore. I think CMake was assuming the `ccache` part of `C=ccache gcc` as the C compiler.

It detects uasm fine and even compiles fine. It just can't identify it as masm. But I think both jwasm and uasm were made to be plug-and-play replacements of masm.